### PR TITLE
Announcement sticky

### PR DESCRIPTION
**Pull request by wangqiang1208**
depends on #62 

Implemented the sticky feature of announcements in this PR.

### CODE REVIEW

**Comment by lowjoel** [comment]
Can you rebase please? sorry :cry:

File: app/controllers/course/announcements_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class Course::AnnouncementsController < Course::ModuleController
   load_and_authorize_resource :announcement, through: :course, class: Course::Announcement.name
 
   def index #:nodoc:
-    @announcements = @announcements.includes(:creator).sorted_by_date
+    @announcements = @announcements.includes(:creator).sorted_by_sticky.sorted_by_date
```
Comment (wangqiang1208) [inline app/controllers/course/announcements_controller.rb]: >_< I think hound settings is missing

**Comment by wangqiang1208** [comment]
@lowjoel, I've rebased this. despite the hound configuring issues, can this be merged  ?

File: app/controllers/course/announcements_controller.rb (hunk 1 above)
Comment (lowjoel) [inline app/controllers/course/announcements_controller.rb]: Is Hound still complaining? It should be much better now.

Comment (wangqiang1208) [inline app/controllers/course/announcements_controller.rb]: Hound it not complain here anymore

**Comment by lowjoel** [comment]
Rebase please

**Comment by wangqiang1208** [comment]
@lowjoel Rebased and checked with rubocop already

File: app/controllers/course/announcements_controller.rb [hunk 2]
```diff
@@ -2,7 +2,8 @@ class Course::AnnouncementsController < Course::ModuleController
   load_and_authorize_resource :announcement, through: :course, class: Course::Announcement.name
 
   def index #:nodoc:
-    @announcements = @announcements.includes(:creator).sorted_by_date.page params[:page]
+    @announcements = @announcements.includes(:creator).
```
Comment (lowjoel) [inline app/controllers/course/announcements_controller.rb]: this old one should be fine... it should be less than 100 chars

Comment (wangqiang1208) [inline app/controllers/course/announcements_controller.rb]: There is a <code>sorted_by_date</code> added

**Pull request merged**